golang/internal.go

```diff
@@ -0,0 +1,244 @@
+				log.Printf("Cache hit: %v\n", entry.UserRequest)
+				completion.Locators = validLocators
+				completion.LocatorType = entry.LocatorType
+				completion.CacheHit = true
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("no cache entry found for user request: %v", userRequest)
+}
+
+// processIdRequest finds elements by analyzing DOM structure and element IDs.
+// Parameters:
+//   - completion: Output structure to populate with results
+//   - userRequest: Natural language description of target element
+//   - dom: Current page DOM representation
+//
+// Returns error if no matching elements are found.
+// Updates completion with token usage and timing metrics.
+func (l *Locatr) processIdRequest(completion *types.LocatrCompletion, userRequest string, dom *types.DOM) error {
+	domChunks, err := l.getRerankedChunks(dom.RootElement.Repr(), userRequest)
+	if err != nil {
+		return err
+	}
+	if len(domChunks) == 0 {
+		return fmt.Errorf("no chunks to process")
+	}
+
+	locatorMap := dom.Metadata.LocatorMap
+	// TODO: Run evals to verify If this approach needs to be changed
+	for _, chunk := range domChunks {
+		idComletion, err := l.options.LLMClient.GetIdCompletion(userRequest, chunk)
+		completion.TimeTaken += idComletion.TimeTaken
+		completion.InputTokens += idComletion.InputTokens
+		completion.OutputTokens += idComletion.OutputTokens
+		completion.TotalTokens += idComletion.TotalTokens
+		if err != nil {
+			log.Printf("couldn't get ID completion: %v\n", err)
+			continue
+		}
+		if idComletion.ErrorMessage != "" {
+			log.Printf("error getting relevant ID: %v\n", idComletion.ErrorMessage)
+			continue
+		}
+		locators := locatorMap[idComletion.Id]
+		if len(locators) == 0 {
+			log.Printf("no locators found for ID: %v\n", idComletion.Id)
+			continue
+		}
+		completion.Locators = locators
+		completion.LocatorType = dom.Metadata.LocatorType
+		return nil
+	}
+	return errors.New("no relevant locator ID found in the DOM")
+}
+
+// processPointRequest finds elements using visual grounding via screenshots.
+// Parameters:
+//   - completion: Output structure to populate with results
+//   - userRequest: Natural language description of target element
+//   - dom: Current page DOM representation
+//
+// Returns error if no matching elements are found.
+// Process includes:
+// 1. Setting viewport size
+// 2. Scrolling to candidate elements
+// 3. Taking screenshots
+// 4. Using LLM to identify click points
+// 5. Converting points to locators
+func (l *Locatr) processPointRequest(completion *types.LocatrCompletion, userRequest string, dom *types.DOM) error {
+	domChunks, err := l.getRerankedChunks(dom.RootElement.Repr(), userRequest)
+	if err != nil {
+		return err
+	}
+	if len(domChunks) == 0 {
+		return fmt.Errorf("no chunks to process")
+	}
+
+	locatorMap := dom.Metadata.LocatorMap
+	for _, chunk := range domChunks {
+		id, err := extractFirstUniqueID(chunk)
+		if err != nil {
+			continue
+		}
+		l.plugin.SetViewportSize(1280, 800)
```

Comment:
nit: let user decide the resolution from a list of supported resolutions by anthropic's computer use.